packages/labs/motion/src/view-transition-update.ts

```diff
@@ -0,0 +1,48 @@
+import {LitElement} from 'lit';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+/** Extend a LitElement to use view transitions (if available) on DOM updates */
+export const ViewTransitionUpdate =
+    <T extends Constructor<LitElement>>(superClass: T) => {
+    class LitElementTransition extends superClass {
+        private lifecycleRenderTransition: ViewTransition;
+  
+        async performUpdate() {
+            if (!(document as DocumentWithTransition).startViewTransition)
+                return await super.performUpdate();
+    
+            if (this.lifecycleRenderTransition)
+                await this.lifecycleRenderTransition.finished;
+    
+            this.lifecycleRenderTransition = (document as DocumentWithTransition).startViewTransition(
+                async () => 
```

Comment:
async not needed. If `super.performUpdate()` were async, it would return a Promise that we just forward.

Reply:
I've added d8f3449121029757fbc52a161fa0073f790b91e9 to removed that `await` and add `return this.updateComplete;` instead, as we need the DOM changes to complete before the promise returns.